File: wep.py
#!/usr/bin/env python

import sys, binascii, md5
import sploit

def show_help():
	print sys.argv[0] + " 'string to convert to four WEP 40 keys and a WEP 104 key'"
	print sys.argv[0] + " -h => shows this help"

def wep_40(string):
	pseed = [0, 0, 0, 0]
	k64 = ["", "", "", ""]
	
	for i in range(0, len(string)):
		pseed[i%4] ^= ord(string[i])
	
	rand_number = pseed[0] | (pseed[1] << 8) | (pseed[2] << 16) | (pseed[3] << 24)
	
	for i in range(0, 4):
		for j in range(0, 5):
			rand_number = (rand_number * 0x343fd + 0x269ec3) & 0xffffffff;
			tmp = (rand_number >> 16) & 0xff;
			k64[i] += binascii.hexlify(chr(tmp)).upper()
		
		print "WEP 40 key" + str(i + 1) + ": " + k64[i]

def pad_to_64(string):
	ret = ""
	rep = 1 + (64 / len(string))
	
	for n in range(0, rep):
		ret += string
	
	return ret[:64]

def wep_104(string):
	string = pad_to_64(string)
	ret = md5.new(string).hexdigest()
	ret = ret.upper()
	
	print "WEP 104 key: " + ret[:26]

def wep_keys():
	string = sploit.bare_words()
	
	wep_40(string)
	wep_104(string)

if __name__ == "__main__":
	try:
		if sys.argv[1] == "-h":
			show_help()
		else:
			wep_keys()
	except IndexError:
		show_help()

File: wep.js
#!/usr/bin/env node

var p = require('path');
var crypto = require('crypto');

var showHelp = function () {
	console.log('./' + p.basename(process.argv[1]) + ' \'string to convert to four WEP 40 keys and a WEP 104 key\'');
	console.log('./' + p.basename(process.argv[1]) + ' -h => shows this help');
}

function binToHex(chr) {
	return ('0' + chr.toString(16)).substr(-2).toUpperCase();
}

var wep40 = function (str) {
	var pseed = [0, 0, 0, 0];
	var k64 = ['', '', '', ''];
	var i, j, tmp;
	
	for (i = 0; i < str.length; i++) {
		pseed[i%4] ^= str.charCodeAt(i);
	}
	
	randNumber = pseed[0] | (pseed[1] << 8) | (pseed[2] << 16) | (pseed[3] << 24);
	
	for (i = 0; i < 4; i++) {
		for (j = 0; j < 5; j++) {
			randNumber = (randNumber * 0x343fd + 0x269ec3) & 0xffffffff;
			tmp = (randNumber >> 16) & 0xff;
			k64[i] += binToHex(tmp);
		}
		
		console.log('WEP 40 key%d: %s', i + 1, k64[i]);
	}
}

var padTo64 = function (str) {
	var i, n, ret = '';
	
	n = 1 + (64 / (str.length));
	
	for (i = 0; i < n; i++)
	{
		ret += str;
	}
	
	return ret.substring(0, 64);
}

var wep104 = function (str) {
	str = padTo64(str);
	var ret = crypto.createHash('md5').update(str).digest('hex').toUpperCase();
	
	console.log('WEP 104 key: %s', ret.substring(0, 26));
}

var wepKeys = function () {
	str = process.argv.slice(2).join(' ');
	console.log('The used string: =>%s<=', str);
	
	wep40(str);
	wep104(str);
}

if (process.argv[2] === '-h' || process.argv.length === 2) {
	showHelp();
} else {
	wepKeys();
}
